src/AutoMapper/Execution/TypeMapPlanBuilder.cs

```diff
@@ -522,6 +522,10 @@ private Expression BuildValueResolverFunc(PropertyMap propertyMap, Expression de
                     );
             }
 
+            valueResolverFunc = _typeMap.Profile.ValueTransformers
+                .Where(vt => vt.IsMatch(propertyMap))
+                .Aggregate(valueResolverFunc, (current, vtConfig) => ToType(ReplaceParameters(vtConfig.TransformerExpression, ToType(current, vtConfig.ValueType)), propertyMap.DestinationPropertyType));
```

Comment:
I wasn't going to make it like IObjectMapper though, more like IValueResolver and ITypeConverter. You can go the expression route, which can be used in LINQ etc. or the runtime route that can get access to ResolutionContext. That would also allow a DI-able version.